pkg/specgen/volumes.go

```diff
@@ -55,149 +55,202 @@ type ImageVolume struct {
 	ReadWrite bool
 }
 
-// GenVolumeMounts parses user input into mounts, volumes and overlay volumes
-func GenVolumeMounts(volumeFlag []string) (map[string]spec.Mount, map[string]*NamedVolume, map[string]*OverlayVolume, error) {
-	mounts := make(map[string]spec.Mount)
-	volumes := make(map[string]*NamedVolume)
-	overlayVolumes := make(map[string]*OverlayVolume)
+type volume struct {
+	src  string
+	dest string
+}
 
+func parseVolume(vol string) ([]volume, []string, error) {
 	volumeFormatErr := errors.New("incorrect volume format, should be [host-dir:]ctr-dir[:option]")
-	for _, vol := range volumeFlag {
-		var (
-			options []string
-			src     string
-			dest    string
-			err     error
-		)
-
-		splitVol := SplitVolumeString(vol)
-		if len(splitVol) > 3 {
-			return nil, nil, nil, fmt.Errorf("%v: %w", vol, volumeFormatErr)
-		}
+	var (
+		err     error
+		volumes []volume
+		options []string
+	)
 
-		src = splitVol[0]
+	splitVol := SplitVolumeString(vol)
+	if len(splitVol) > 3 {
+		return nil, nil, fmt.Errorf("%v: %w", vol, volumeFormatErr)
+	}
 
-		// Support relative paths beginning with ./
-		if strings.HasPrefix(src, "./") {
-			path, err := filepath.EvalSymlinks(src)
-			if err != nil {
-				return nil, nil, nil, err
-			}
-			src, err = filepath.Abs(path)
-			if err != nil {
-				return nil, nil, nil, err
+	srcs, err := filepath.Glob(splitVol[0])
```

Comment:
I know we've had CVEs in the past when we supported this with `podman cp`. I'm a little worried about allowing globs blindly.

Reply:
Also, isn't `*` a valid character to have in a filename? Is that going to be a problem, or will glob expansion handle that without a problem?

Reply:
ls /tmp/foo\* will match /tmp/foo\*

Reply:
I am not sure if there is a risk of filepath.Glob(), it will basically just mount every file that matches the glob onto itself
inside of the container. If you had a sympolic link from one of the globs, I don't think filepath.Glob will resolve it, but when you end up with -v /tmp/foobar:/tmp/foobar which could be a symbolic link, but this should not be an issue.